### PR TITLE
usage/doc synchronization

### PR DESCRIPTION
**Pull request by smikes**
As a first approach to the problem laid out in #7082, this PR aims to bring the short help displayed on `npm <subcommand> --help` and the `SYNOPSIS` section of the `cli/npm-<subcommand>.md` into harmony.

There is a list of all current published `npm` subcommands, and both the `--help` usage message and `SYNOPSIS` text, on tab `UsageSync` of my spreadsheet, here - https://docs.google.com/spreadsheets/d/1r-Gn2mEqPDwDVwIWYxP-ChxFQFVtrju5HEjruEa0tyQ/edit#gid=1285621165

There are a number of variations which need to be handled -- e.g., some subcommands `--help` launches a page in the pager/browser; etc. I'm opening this PR now so discussion about those things can happen.

This PR is based against `npm/multi-stage` and will be targeted for `npm@3` or later.

Initial changes:

> whoami: add --registry to usage
> remove npm-rm.md, redundant to npm-uninstall.md

### CODE REVIEW

**Comment by smikes** [comment]
As the initial change, I noted that `npm-rm.md` and `npm-uninstall.md` document the same command.  See  <https://docs.npmjs.com/cli/rm> and <https://docs.npmjs.com/cli/uninstall>.

Since `uninstall` is longer, I have kept it and removed `npm-rm.md`.  An alias or redirect should be created on the docs website.

**Comment by othiym23** [comment]
We should probably figure out how to deal with those redirects on the client side, @smikes. I need to read up on how `mandb` deals with multiple commands being pointed at the same page, but npm should do something to be more consistent. I think for now, nuking / merging duplicate entries is a sensible plan. @isaacs might have had a reason once for having `rm` and `uninstall` as separate pages, but I no longer know what it is.

**Comment by othiym23** [comment]
See also #7600 / #7601. What are next steps here? Having PRs sitting around the queue with `NOT READY` in the name makes me twitchy. ;)

**Comment by smikes** [comment]
How about I split off the commit for combining rm and uninstall into its own PR;

and then tomorrow morning I can finish the first pass on synchronizing the --usage messages with the SYNOPSIS sections and then you can review it.

**Comment by smikes** [comment]
I am ready for review of this

File: doc/cli/npm-completion.md [hunk 1]
```diff
@@ -3,7 +3,10 @@ npm-completion(1) -- Tab Completion for npm
 
 ## SYNOPSIS
 
-    . <(npm completion)
+    npm completion >> ~/.bashrc
+    npm completion >> ~/.zshrc
+
+    source <(npm completion)
```
Comment (othiym23) [inline doc/cli/npm-completion.md]: The way this is written is a little confusing. Maybe move the `npm completion >> ~ /.{z,ba}shrc` stuff into a section down below?

File: doc/cli/npm-config.md [hunk 2]
```diff
@@ -8,7 +8,6 @@ npm-config(1) -- Manage the npm configuration files
     npm config delete <key>
     npm config list
     npm config edit
-    npm c [set|get|delete|list]
```
Comment (othiym23) [inline doc/cli/npm-config.md]: good call.

File: doc/cli/npm-edit.md [hunk 3]
```diff
@@ -3,7 +3,7 @@ npm-edit(1) -- Edit an installed package
 
 ## SYNOPSIS
 
-    npm edit <name>[@<version>]
+    npm edit <pkg>
```
Comment (othiym23) [inline doc/cli/npm-edit.md]: For this command, and a few others, making clear that you can fetch a specific version of the package in the command is useful, inasmuch as most people wouldn't even think that was a possibility if they came to the manpage cold.

File: doc/cli/npm-install.md [hunk 4]
```diff
@@ -3,15 +3,23 @@ npm-install(1) -- Install a package
...
+    npm install <tarball file>
+    npm install <tarball url>
+    npm install <folder>
+
+    alias: npm i
+    common options: [--save|--save-dev|--save-optional] [--save-exact]
+
+    Can specify one or more: npm install ./foo.tgz bar@stable /some/folder
+
+    If no argument is supplied and ./npm-shrinkwrap.json is present,
```
Comment (othiym23) [inline doc/cli/npm-install.md]: This section feels out of place in the `SYNOPSIS`. It's OK to draw people's attention to it with a callout, but it should be in the body of the `DESCRIPTION` with the rest of the docs.

File: doc/cli/npm-logout.md [hunk 5]
```diff
@@ -3,7 +3,7 @@ npm-logout(1) -- Log out of the registry
 
 ## SYNOPSIS
 
-    npm logout [--registry=url] [--scope=@orgname]
+    npm logout [--registry=<url>] [--scope=@<scope>]
```
Comment (othiym23) [inline doc/cli/npm-logout.md]: `--scope=<@scope>` – this is a point of some contention within the npm team, as @zeke can attest, but my feeling is that the `@` is part of the scope name.

File: doc/cli/npm-outdated.md [hunk 6]
```diff
@@ -3,7 +3,7 @@ npm-outdated(1) -- Check for outdated packages
 
 ## SYNOPSIS
 
-    npm outdated [<name> [<name> ...]]
+    npm outdated [[@<scope>/]<pkg> ...]
```
Comment (othiym23) [inline doc/cli/npm-outdated.md]: Same point about `<@scope>` – I think this one is probably on me originally, in that I've changed my mind about what makes the most sense. So I can make this change if it seems too exasperating to you.

File: doc/cli/npm-install.md (hunk 4 above)
Comment (smikes) [inline doc/cli/npm-install.md]: OK, I can pull it.  That means pulling it from the `usage` too, though -- is that desirable?

File: doc/cli/npm-outdated.md (hunk 6 above)
Comment (smikes) [inline doc/cli/npm-outdated.md]: perl one-liner. already done.

**Comment by smikes** [comment]
ready for additional review

File: doc/cli/npm-completion.md [hunk 7]
```diff
@@ -3,7 +3,7 @@ npm-completion(1) -- Tab Completion for npm
 
 ## SYNOPSIS
 
-    . <(npm completion)
+    source <(npm completion)
```
Comment (zeke) [inline doc/cli/npm-completion.md]: yes!

File: doc/cli/npm-ls.md [hunk 8]
```diff
@@ -3,10 +3,9 @@ npm-ls(1) -- List installed packages
 
 ## SYNOPSIS
 
-    npm list [[@<scope>/]<pkg> ...]
-    npm ls [[@<scope>/]<pkg> ...]
-    npm la [[@<scope>/]<pkg> ...]
-    npm ll [[@<scope>/]<pkg> ...]
+    npm ls [[<@scope>/]<pkg> ...]
+
+    aliases: list, la, ll
```
Comment (othiym23) [inline doc/cli/npm-ls.md]: It's worth noting at some point that `npm ll` and `npm la` do something very different from `npm ls` (try them!), but that doesn't need to be part of this commit.

**Comment by othiym23** [comment]
This LGTM, but I leave it to @iarna to merge changes into `multi-stage` when she thinks they're ready.

**Comment by iarna** [comment]
Landed as c20e798 in https://github.com/npm/npm/tree/multi-stage

**Pull request closed without merging**